src/views/dashboard/modules/data-view/components/data-grid.vue

```diff
@@ -28,25 +50,80 @@ a-card(:bordered="false")
     'show-page-size': true,
   }
 
-  const gridColumn = computed(() => {
+  // replace '.' with '-' to make it a valid data-index
+  // useful when '.' is used in column name, such as `SELECT 1.1`, `SELECT 1 AS a.b`
+  function columnNameToDataIndex(columnName: string) {
+    return columnName.replace(/\./gi, '-')
+  }
+
+  const timeColumnNames = computed(() => {
+    return props.data.records.schema.column_schemas
+      .filter((column: any) => dateTypes.includes(column.data_type))
+      .map((column: any) => column.name)
+  })
+
+  const gridColumns = computed(() => {
     const { schema } = props.data.records
     if (!schema) return []
-    return schema.column_schemas.map((column: any) => {
-      return {
-        title: column.name,
-        dataIndex: column.name.replace(/\./gi, '-'),
-      }
-    })
-  })
 
-  const gridData = computed(() => {
-    return props.data.records.rows.map((row: any) => {
-      const tempRow: any = {}
-      row.forEach((item: any, index: number) => {
-        const columnName = props.data.records.schema.column_schemas[index].name.replace(/\./gi, '-')
-        tempRow[columnName] = item
+    // use sort to make time columns display on the left first
+    return schema.column_schemas
+      .map((column: any) => {
+        return {
+          title: column.name,
+          dataIndex: columnNameToDataIndex(column.name),
+        }
+      })
+      .sort((a: any, b: any) => {
+        if (timeColumnNames.value.includes(a.title)) {
+          return -1
```

Comment:
```suggestion
          if (timeColumnNames.value.includes(b.title)) {
            return 0
          }
          return -1
```

Reply:
haha, I forgot to check the logic

Reply:
To keep the time columns to be the original order.

Reply:
```ts
sort((a: any, b: any) => {
        return timeColumnNames.value.includes(b.title) - timeColumnNames.value.includes(a.title)
})
```
This would be shorter.
I have tested the logic